core/src/main/java/de/bwaldvogel/mongo/backend/AbstractMongoDatabase.java

```diff
@@ -275,11 +244,11 @@ private static QueryParameters toQueryParameters(Document query) {
         querySelector.put("$query", query.getOrDefault("filter", new Document()));
         querySelector.put("$orderby", query.get("sort"));
 
-        Document projection = (Document) query.get("projection");
-        return new QueryParameters(querySelector, numberToSkip, numberToReturn, batchSize, projection);
+        return new QueryParameters(querySelector, numberToSkip, numberToReturn, batchSize,
+            (Document) query.get("projection"));
```

Comment:
is this inlining part of the revert or why was it done?

Reply:
reverted